plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -682,6 +682,13 @@ public WriteMapping toWriteMapping(ConnectorSession session, Type type)
         return aggregateFunctionRewriter.rewrite(session, aggregate, assignments);
     }
 
+    @Override
+    public boolean supportsAggregationPushdown(ConnectorSession session, JdbcTableHandle table, List<AggregateFunction> aggregates, Map<String, ColumnHandle> assignments, List<List<ColumnHandle>> groupingSets)
+    {
+        // Postgres sorts textual types differently compared to Trino so we cannot safely pushdown any aggregations which take a text type as an input or as part of grouping set
```

Comment:
if it's about sorting (as we believe it is), limitation doesn't apply to GROUPING SETs. 
Also it applies to some aggregation functions (`min`, `max`), but not others `count`, or `count(DISTINCT)`

Reply:
Yes. The part about difference between aggregation functions came up before but I couldn't find a nice way to model it (other than introduce a field to AggregateFunction and having each of them declare if they depend on case-sensitivity for correctness).

For the grouping set - thanks for catching. Postgres indeed doesn't need the grouping set check - only the aggregate function check.

Reply:
Re how to model - since this is function specific, it could be handled by `io.trino.plugin.jdbc.expression.AggregateFunctionRule#rewrite` returning `empty` or not.

Reply:
That's what I tried initially but it's a bit errorprone since each module has their own rewrites. I can try to move as many to base-jdbc as possible in a preparatory commit.

Reply:
Also, to confirm, you mean that the grouping sets condition can be handled in `supportsAggregation` like now.
And instead of handling the functions in `supportsAggregation` I should handle them in the rewrite rules?

Reply:
> That's what I tried initially but it's a bit errorprone since each module has their own rewrites. I can try to move as many to base-jdbc as possible in a preparatory commit.

That's unlikely to work. We have separate rewrites because there were some differences.

> Also, to confirm, you mean that the grouping sets condition can be handled in supportsAggregation like now.

i didn't read that part, so "maybe"

> And instead of handling the functions in supportsAggregation I should handle them in the rewrite rules?

that's my intuition

Reply:
I implemented the approach you suggested - it works and thankfully the only case-sensitive rewrites live in base-jdbc already.